packages/taste-test/src/utils.ts

```diff
@@ -26,3 +26,62 @@ export const findCoveringNode = (utxos: UTxO[], userKey: string) =>
       (datum.next == null || userKey < datum.next)
     );
   });
+
+/**
+ * Searches through a list of UTXOs to find a node owned by the user, identified by a specific key.
+ *
+ * @param {UTxO[]} utxos - An array of unspent transaction outputs (UTXOs).
+ * @param {string} userKey - The unique identifier key for the user, used to find a specific node in the UTXOs.
+ * @returns {UTxO | undefined} - Returns the UTXO that contains the user's node if found, otherwise returns undefined.
+ *
+ * This function iterates through the provided `utxos`, attempting to match the `userKey` with a key within the datum of each UTXO.
+ * If a match is found, it indicates that the UTXO belongs to the user's node, and this UTXO is returned.
+ * If no matching node is found in the list of UTXOs, the function returns undefined, indicating the user's node was not found.
+ */
+export const findOwnNode = (utxos: UTxO[], userKey: string) =>
+  utxos.find((value) => {
+    if (!value.datum) {
+      return false;
+    }
+
+    const nodeData = Data.from(value.datum, SetNode);
+    return nodeData.key == userKey;
+  });
+
+/**
+ * Searches through a list of UTXOs to find a node owned by the user, identified by a specific key, and return the next reference (previous to the user's node).
+ *
+ * @param {UTxO[]} utxos - An array of unspent transaction outputs (UTXOs).
+ * @param {string} userKey - The unique identifier key for the user, used to find a specific node in the UTXOs.
+ * @returns {UTxO | undefined} - Returns the UTXO that contains the previous node of the user's node if found, otherwise returns undefined.
+ *
+ * This function iterates through the provided `utxos`, attempting to match the `userKey` with a key within the datum of each UTXO.
+ * If a match is found, it indicates that the UTXO belongs to the user's node, and the UTXO referenced as the previous node is returned.
+ * If no matching node is found in the list of UTXOs, the function returns undefined, indicating the previous node of the user's node was not found.
+ */
+export const findPrevNode = (utxos: UTxO[], userKey: string) =>
+  utxos.find((value) => {
+    if (!value.datum) {
+      return false;
+    }
+
+    const datum = Data.from(value.datum, SetNode);
+    return datum.next !== null && datum.next == userKey;
```

Comment:
We can simplify this condition here as `userKey` can not be `null` according to the types

```suggestion
    return datum.next === userKey;
```